test/metric/diskio.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric
+
+import (
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+)
+
+type DiskIOMetricValueFetcher struct {
+	baseMetricValueFetcher
+}
+
+var _ MetricValueFetcher = (*DiskIOMetricValueFetcher)(nil)
+
+func (f *DiskIOMetricValueFetcher) Fetch(namespace, metricName string, stat Statistics) (MetricValues, error) {
+	dims := f.getMetricSpecificDimensions()
+	values, err := f.fetch(namespace, metricName, dims, stat)
+	if err != nil {
+		log.Printf("Error while fetching metric value for %s: %v", metricName, err)
+	}
+	return values, err
+}
+
+func (f *DiskIOMetricValueFetcher) isApplicable(metricName string) bool {
+	diskIOSupportedMetric := f.getPluginSupportedMetric()
+	_, exists := diskIOSupportedMetric[metricName]
+	return exists
+}
+
+func (f *DiskIOMetricValueFetcher) getPluginSupportedMetric() map[string]struct{} {
+	// DiskIO Supported Metrics
```

Comment:
Good raise up question. However, it would be depends on the metrics sent. For example, network receive bytes can be up more than 1,000,000,000 Bytes while the delta value is around 10,000.

Reply:
So which value are we emitting? Kaushik's point is that we have some plugins that rely on the delta processor to publish incremental data. If you look at the source of netstat data, it scrapes from a proc file that always increases, which requires us to compute the delta before submitting. That means that we have to verify that behavior when we do benchmark testing to assert that we get a reasonable value for the delta, and not just the raw number of network bytes that are sent.